CRM/Core/BAO/CustomValueTable.php

```diff
@@ -217,6 +217,9 @@ public static function create(&$customParams) {
               break;
 
             default:
+              if ($value == NULL) {
```

Comment:
Can you explain this line (or better yet add code comments)

Reply:
When you pass NULL value to custom field of type float then you get fatal error with 
<code>
Jan 16 15:08:52  [info] $Fatal Error Details = Array
(
    [message] => One of parameters  (value: ) is not of the type Float
    [code] => 
)


Jan 16 15:08:52  [info] $backTrace = #0 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Core/Error.php(378): CRM_Core_Error::backtrace("backTrace", TRUE)
#1 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Utils/Type.php(504): CRM_Core_Error::fatal("One of parameters  (value: ) is not of the type Float")
#2 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Core/DAO.php(1424): CRM_Utils_Type::validate("", "Float")
#3 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Core/DAO.php(1301): CRM_Core_DAO::composeQuery("UPDATE civicrm_value_food_preference_2  SET soup_selection_4 = %2,number_fiel...", (Array:3), TRUE)
#4 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Core/BAO/CustomValueTable.php(255): CRM_Core_DAO::executeQuery("UPDATE civicrm_value_food_preference_2  SET soup_selection_4 = %2,number_fiel...", (Array:3))
#5 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Core/BAO/CustomValueTable.php(363): CRM_Core_BAO_CustomValueTable::create((Array:1))
#6 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Event/BAO/Participant.php(228): CRM_Core_BAO_CustomValueTable::store((Array:2), "civicrm_participant", "54")
#7 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Event/Form/Participant.php(1322): CRM_Event_BAO_Participant::create((Array:26))
#8 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Event/Form/Participant.php(934): CRM_Event_Form_Participant->submit((Array:18))
#9 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Core/Form.php(447): CRM_Event_Form_Participant->postProcess()
#10 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Core/QuickForm/Action/Upload.php(169): CRM_Core_Form->mainProcess()
#11 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Core/QuickForm/Action/Upload.php(136): CRM_Core_QuickForm_Action_Upload->realPerform(Object(CRM_Event_Form_Participant), "upload")
#12 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/packages/HTML/QuickForm/Controller.php(203): CRM_Core_QuickForm_Action_Upload->perform(Object(CRM_Event_Form_Participant), "upload")
#13 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/packages/HTML/QuickForm/Page.php(103): HTML_QuickForm_Controller->handle(Object(CRM_Event_Form_Participant), "upload")
#14 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Core/Controller.php(351): HTML_QuickForm_Page->handle("upload")
#15 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Event/Page/Tab.php(118): CRM_Core_Controller->run()
#16 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Event/Page/Tab.php(175): CRM_Event_Page_Tab->edit()
#17 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Core/Invoke.php(309): CRM_Event_Page_Tab->run((Array:4), NULL)
#18 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Core/Invoke.php(84): CRM_Core_Invoke::runItem((Array:12))
#19 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/CRM/Core/Invoke.php(52): CRM_Core_Invoke::_invoke((Array:4))
#20 /Users/kapil/Sites/drupal-7.56/sites/all/modules/civicrm/drupal/civicrm.module(448): CRM_Core_Invoke::invoke((Array:4))
#21 [internal function](): civicrm_invoke("contact", "view", "participant")
#22 /Users/kapil/Sites/drupal-7.56/includes/menu.inc(527): call_user_func_array("civicrm_invoke", (Array:3))
#23 /Users/kapil/Sites/drupal-7.56/index.php(21): menu_execute_active_handler()
#24 {main}
</code>

Therefore i checked If Value == NULL and explicitly assigned 'Timestamp' to$type to bypass the error. This has been followed elsewhere in code(I know its a bad idea :( ).

eg: https://github.com/pradpnayak/civicrm-core/blob/76487bb9de979deef99205295b5a19bba3613b7c/CRM/Core/BAO/CustomValueTable.php#L158